docs: add how to start a server/client (#62)
decentraland/rpc-rust: README.md

```

This include should be done in the `src/lib.rs` so the entire crate handle types correctly, otherwise it will treat every include as different types.

### Server Side

```rust
use dcl_rpc::{transports::web_socket::{WebSocketServer, WebSocketTransport}, server::{RpcServer, RpcServerPort}, service_module_definition::{Definition, ServiceModuleDefinition, CommonPayload}};

// Define the IP and Port where the WebSocket Server will run
let ws_server = WebSocketServer::new("localhost:8080");
// Start listening on that IP:PORT
let mut connection_listener = ws_server.listen().await.unwrap();

// Add here any data that the server needs to solve the messages, for example db.
let ctx = MyExampleContext {
    hardcoded_database: create_db(),
};

let mut server = RpcServer::create(ctx);
server.set_handler(|port: &mut RpcServerPort<MyExampleContext>| {
  // The EchoServiceRegistration will be autogenerated, so you'll need to define the echo_service, which will have all the behaviors of your service. Following the example, it'll have the logic for the `hello` message.
  EchoServiceRegistration::register_service(port, echo_service::EchoService {})
});

> The WebSocket Server ?) listens for incoming connections, when a connection is established, it creates a new WebSocketTransport with that connection and attaches it to the server event sender. The loop continues to listen for incoming connections and attach transports until it is stopped.

Reviewer: ```suggestion
// The WebSocket Server ?) listens for incoming connections, when a connection is established, it creates a new WebSocketTransport with that connection and attaches it to the server event sender. The loop continues to listen for incoming connections and attach transports until it is stopped.
```

> // The WebSocket Server ?) listens for incoming connections, when a connection is established, it creates a new WebSocketTransport with that connection and attaches it to the server event sender. The loop continues to listen for incoming connections and attach transports until it is stopped.

Reviewer 2: ```suggestion
// The WebSocket Server listens for incoming connections, when a connection is established, it creates a new WebSocketTransport with that connection and attaches it to the server event sender. The loop continues to listen for incoming connections and attach transports until it is stopped.
```

### Run the multi language integration example: RPC Client in Typescript and RPC Server in Rust

`just run-multilang`

## Usage

### Import

```toml
[dependencies]
dcl-rpc = "latest"
````

### Protobuf

Create a file `app.proto` to define the messages that will be used, for example:

```proto
syntax = "proto3";
package decentraland.echo;

message Text {
  string say_something = 1;
}

service EchoService {
  rpc Hello(Text) returns (Text) {}
}
```

Then, define a `build.rs` file to build the types of the message:

```rust

>   extern crate prost_build;

Reviewer: I believe this line is not required
```suggestion

```

  use std::io::Result;

  fn main() -> Result<()> {
      // Tell Cargo that if the given file changes, to rerun this build script.
      println!("cargo:rerun-if-changed=src/echo.proto");

>       prost_build::compile_protos(&["src/echo.proto"], &["src"])?;

Reviewer: what do you say if we add the statement to generate client and server traits and related code, for that we should add the `dcl-rpc-codegen` as a _build-dependecy_ and change this `build.rs`

and then change this line:
```suggestion
      let mut config = prost_build::Config::new();
    config.service_generator(Box::new(dcl_rpc_codegen::RPCServiceGenerator::new()));
    config.compile_protos(&["src/echo.proto"], &["src"])?;
```

      Ok(())
  }
```

That code will build the necessary modules to support the custom messages defined in the protobuf and use them in Rust.

> To import them you will need to add:

Reviewer: I would add that this include should be done in the `lib.rs` so the entire crate handle types correctly, otherwise it will treat every include as different types (not sure what's the reason)

```rust
include!(concat!(env!("OUT_DIR"), "/decentraland.echo.rs"));
```

### Server Side

```rust
use dcl_rpc::{transports::web_socket::{WebSocketServer, WebSocketTransport}, server::{RpcServer, RpcServerPort}, service_module_definition::{Definition, ServiceModuleDefinition, CommonPayload}};

// Define the IP and Port where the WebSocket Server will run
let ws_server = WebSocketServer::new("localhost:8080");
// Start listening on that IP:PORT
let mut connection_listener = ws_server.listen().await.unwrap();

// Add here any data that the server needs to solve the messages, for example db.
let ctx = MyExampleContext {
    hardcoded_database: create_db(),
};

// Create the Server with the context needed

> let mut server = RpcServer::create(ctx);

Reviewer 2: I think this is repeated code. Isn't it? (check line 132)

// Define the handlers for the messages defined in the proto file

> server.set_handler(|port: &mut RpcServerPort<()>| {

Reviewer 2: Same with this block, check line 133

    let mut module = ServiceModuleDefinition::new();
    // Define the behaviour of Hello Message as an echo server
    module.add_unary("Hello", |payload, ctx| { Box::pin(async {payload}) });
    port.register_module("EchoService".to_string(), module);
});

let mut server = RpcServer::create(ctx);
server.set_handler(|port: &mut RpcServerPort<MyExampleContext>| {

>   // The EchoServiceCodeGen will be autogenerated, you need to define the echo_service

Reviewer 2: I think I'd like to know more explicitly what "define the echo_service" means: 
```suggestion
// The EchoServiceCodeGen will be autogenerated, so you'll need to define the echo_service, which will have all the behaviors of your service. Following the example, it'll have the logic for the `hello` message.
```

  EchoServiceCodeGen::register_service(port, echo_service::EchoService {})
});

> // It has to use the server events sender to attach transport because it has to wait for client connections

Reviewer 2: If I read the comment I don't understand what this block of code does. I think I'd like to read sth like: 
```suggestion
The WebSocket Server ?) listens for incoming connections, when a connection is established, it creates a new WebSocketTransport with that connection and attaches it to the server event sender. The loop continues to listen for incoming connections and attach transports until it is stopped.
```

// and keep waiting for new ones
let server_events_sender = server.get_server_events_sender();
tokio::spawn(async move {
    while let Some(Ok(connection)) = connection_listener.recv().await {
        let transport = WebSocketTransport::new(connection);
        match server_events_sender.send_attach_transport(transport) {
            Ok(_) => {
                println!("> RpcServer > transport attached successfully");
            }
            Err(_) => {
                println!("> RpcServer > unable to attach transport");
                panic!()
            }
        }
    }
});

server.run().await;
```

Implement the trait for your service

```rust

#[async_trait::async_trait]

> impl EchoServiceInterface<MyExampleContext> for EchoService {

Reviewer: After `rpc_codegen` changes, it will be named `SharedEchoService` instead of `EchoServiceInterface`
```suggestion
impl SharedEchoService<MyExampleContext> for EchoService {
```

Also, I think we should include the `use` statement for that trait.
